File: py/classes.py
import random


class Player:
    def __init__(self, name, class_, FOR, END, HAB, CHA, INT, INI, ATT, PAR, TIR, NA, PV):
        self.name = name
        self.class_ = class_
        self.FOR = FOR
        self.END = END
        self.HAB = HAB
        self.CHA = CHA
        self.INT = INT
        self.INI = INI
        self.ATT = ATT
        self.PAR = PAR
        self.TIR = TIR
        self.NA = NA
        self.PV = PV

    def roll(self, dice_value):
        self.dices = [4, 6, 8, 10, 20, 100]
        self.dice_value = dice_value

        if self.dice_value in self.dices:
            self.dice_result = random.randint(1, self.dice_value)
        return self.dice_result

    def introduce(self):
        print(f'{self.name} - {self.class_} \nFor: {self.FOR} \nEnd: {self.END} \nHab: {self.HAB} \nCha: {self.CHA} \nInt: {self.INT} \nIni: {self.INI} \nAtt: {self.ATT} \nPar: {self.PAR} \nTir: {self.TIR} \nNa: {self.NA} \nPv: {self.PV}')

    def equipment(self, right_hand, left_hand, dmg, prot):
        self.right_hand = right_hand
        self.left_hand = left_hand
        self.dmg = dmg
        self.prot = prot

        self.dmg = self.dmg + self.roll(8)
        self.prot = self.prot + self.roll(4)
        return self.dmg, self.prot

    def attack(self):
        self.roll(20)
        print(f'{self.name} tente d\'attaquer: ATT {self.ATT}')
        if self.dice_result <= self.ATT:
            print(f'{self.dice_result} réussi')
            self.success = True
        else:
            print(f'{self.dice_result} échec')
            self.success = False

        return self.success, self.dice_result

    def pary(self):
        self.roll(20)
        print(f'{self.name} tente de parer: PAR {self.PAR}')
        if self.dice_result <= self.PAR:
            print(f'{self.dice_result} réussi')
            self.success = True
        else:
            print(f'{self.dice_result} échec')
            self.success = True
        return self.success, self.dice_result

    def dodge(self):
        self.roll(20)
        print(f'{self.name} tente d\'esquiver: HAB {self.HAB}')
        if self.dice_result <= self.HAB:
            print(f'{self.dice_result} réussi')
            self.success = True
        else:
            print(f'{self.dice_result} échec')
            self.success = True
        return self.success, self.dice_result

File: classes.js
class Character {
    constructor(name, class_, FOR, END, HAB, CHA, INT, INI, ATT, PAR, TIR, NA, PV ) {
        this.name = name;
        this.class_ = class_;
        this.FOR = FOR;
        this.END = END;
        this.HAB = HAB;
        this.CHA = CHA;
        this.INT = INT;
        this.INI = INI;
        this.ATT = ATT;
        this.PAR = PAR;
        this.TIR = TIR;
        this.NA = NA;
        this.PV = PV;
    }

    roll(dice_value) {
        var dices = [4, 6, 8, 10, 20, 100];
      
        if (dices.includes(dice_value)) {
          this.result = Math.floor(Math.random() * dice_value) + 1;
          this.value = "1d" + dice_value;

          this.success = this.result + ' réussi';
          this.success = this.success.fontcolor("green");
  
          this.failure = this.result + " échec";
          this.failure = this.failure.fontcolor("red");

          return this.result;
        } else {
          console.log("Enter a valid number: (4, 6, 8, 10, 20, 100)");
        }
    }

    introduce() {
        console.log(this.name, '-' ,this.class_ , '\n' ,
        "For: ", this.FOR, '\n',
        "End: ", this.END, '\n',
        "Hab: ", this.HAB, '\n',
        "Cha: ", this.CHA, '\n',
        "Int: ", this.INT, '\n',
        "Ini: ", this.INI, '\n',
        "Att: ", this.ATT, '\n',
        "Par: ", this.PAR, '\n',
        "Tir: ", this.TIR, '\n',
        "Na: ", this.NA, '\n',
        "Pv: ", this.PV);
    }

    equipment(weapon, dmg, prot) {
        this.weapon = weapon
        this.dmg = dmg
        this.prot = prot

        this.dmg = this.dmg + this.roll(8)
        this.prot = this.prot + this.roll(4)
        return this.dmg, this.prot
    }

    attack() {
        this.roll(20);
        console.log(this.name + " tente d'attaquer: " + "ATT " + this.ATT)
        if (this.result <= this.ATT) {
            console.log(this.result + ' réussi');
        } else {
            console.log(this.result + ' échec');
        }
    }

    pary() {
        this.roll(20)
        console.log(this.name + " tente de parer: " + "PAR " + this.PAR)
        if (this.result <= this.PAR) {
            console.log(this.result + ' réussi');
        } else {
            console.log(this.result + ' échec');
        }
    }

    dodge() {
        this.roll(20)
        console.log(this.name + " tente d'esquiver: " + "HAB " + this.HAB)
        if (this.result <= this.HAB) {
            console.log(this.result + ' réussi');
        } else {
            console.log(this.result + ' réussi');
        }
    }
}